src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight.kt

```diff
@@ -501,18 +522,39 @@ class ElytraFlight : Module() {
     }
 
     override fun onUpdate() {
+        val oldpathfind = pathfind
+        pathfind = MODULE_MANAGER.isModuleEnabled(AutoWalk::class.java) && MODULE_MANAGER.getModuleT(AutoWalk::class.java).mode.value == AutoWalk.AutoWalkMode.PATHFIND
+        if (oldpathfind == false && pathfind == true) {
+            effectiveMode = ElytraFlightMode.CONTROL
+            Astarpathfinder.init()
+        }
+        if (pathfind == false) {
+            effectiveMode = mode.value
+            if (oldpathfind == true)
+                Astarpathfinder.turnOff()
+        }
+
```

Comment:
Should this really be running when the game is still loading?

Reply:
I could solve this by using a settingslistener on the mode setting from AutoWalk and call some "updatePathfindState" function from autowalk.onenable.  ill push my idea you can have a look at the onenable/disable functions of both modules